src/derive_utils.rs

```diff
@@ -10,106 +10,242 @@ use crate::exceptions::PyTypeError;
 use crate::instance::PyNativeType;
 use crate::pyclass::PyClass;
 use crate::types::{PyAny, PyDict, PyModule, PyString, PyTuple};
-use crate::{ffi, GILPool, IntoPy, PyCell, Python};
+use crate::{ffi, GILPool, PyCell, Python};
 use std::cell::UnsafeCell;
 
-/// Description of a python parameter; used for `parse_args()`.
 #[derive(Debug)]
-pub struct ParamDescription {
-    /// The name of the parameter.
+pub struct KeywordOnlyParameterDescription {
     pub name: &'static str,
-    /// Whether the parameter is optional.
-    pub is_optional: bool,
-    /// Whether the parameter is optional.
-    pub kw_only: bool,
+    pub required: bool,
 }
 
-/// Parse argument list
-///
-/// * fname:  Name of the current function
-/// * params: Declared parameters of the function
-/// * args:   Positional arguments
-/// * kwargs: Keyword arguments
-/// * output: Output array that receives the arguments.
-///           Must have same length as `params` and must be initialized to `None`.
```

Comment:
How about retrieving some of these comments?

Reply:
Good idea.